src/Command/ImportADSCommand.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+
+namespace Adshares\AdsManager\Command;
+
+use Adshares\AdsManager\AdsImporter\Exception\AdsClientException;
+use Adshares\AdsManager\AdsImporter\Importer;
+use Adshares\AdsManager\AdsImporter\ImporterResult;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ImportADSCommand extends ContainerAwareCommand
+{
+    private $adsImporter;
+
+    public function __construct(Importer $adsImporter)
+    {
+        $this->adsImporter = $adsImporter;
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setName('ads:import')
+            ->setDescription('Importing data from ADS client');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): void
+    {
+        try {
+            /** @var ImporterResult $result */
+            $result = $this->adsImporter->import();
+        } catch (AdsClientException $ex) {
+            sleep(3); // retry for new block
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this call to "sleep". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS2964)

Reply:
If ADS client returns `not available block` exception we can wait a few seconds an try again. Other option is to wait for the next running (32s - dev, 512s - prod). 
This situation occurs only when a new block it's being initialized.